### PR TITLE
fix:[CDS-71319]: Add exception when one is missing inside the error response data

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- An NPE is regularly thrown when a delegate task fails during the validation phase. 
- In general, a validation failure does not need to be related to an exception, but from a notify event listener a failure always needs to be related to one.
- Without a clear exception in that case, the git operation fails without raising an alert, and without safely updating the change set status and git commit.

Screenshot 2023-08-30 at 16 00 31

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52271)

### CODE REVIEW

**Comment by Fernando-Dourado** [comment]
trigger uts

File: 400-rest/src/main/java/software/wings/service/impl/yaml/GitCommandCallback.java [hunk 1]
```diff
@@ -440,8 +441,8 @@ public void notify(Map<String, Supplier<ResponseData>> response) {
       try {
         ResponseData responseData = responseDataSupplier.get();
         notify(responseData);
-      } catch (
-          NoAvailableDelegatesException | NoInstalledDelegatesException | NoEligibleDelegatesInAccountException e) {
+      } catch (NoAvailableDelegatesException | NoInstalledDelegatesException | NoEligibleDelegatesInAccountException
```
Comment (rafaelmart1ns) [inline 400-rest/src/main/java/software/wings/service/impl/yaml/GitCommandCallback.java:444]: what do you think about treating the messages separately?

File: 400-rest/src/main/java/io/harness/iterator/FailDelegateTaskIteratorHelper.java [hunk 2]
```diff
@@ -189,7 +190,11 @@ public void endTasks(
                 DelegateTaskResponse.builder()
                     .accountId(delegateTasks.get(taskId).getAccountId())
                     .responseCode(DelegateTaskResponse.ResponseCode.FAILED)
-                    .response(ErrorNotifyResponseData.builder().expired(true).errorMessage(errorMessage).build())
+                    .response(ErrorNotifyResponseData.builder()
+                                  .expired(true)
+                                  .exception(new DelegateIteratorValidationException(errorMessage))
+                                  .errorMessage(errorMessage)
+                                  .build())
```
Comment (jennyjames21) [inline 400-rest/src/main/java/io/harness/iterator/FailDelegateTaskIteratorHelper.java:197]: this is gonna throw exceptions for other task types, we explicitly removed it for the log cost purpose.
Instead rely on error message,.

**Review by jennyjames21 — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/io/harness/iterator/FailDelegateTaskIteratorHelper.java (hunk 2 above)
Comment (Fernando-Dourado) [reply 400-rest/src/main/java/io/harness/iterator/FailDelegateTaskIteratorHelper.java:197]: Now I see clearly what you told me during our last call, let me change it.

**Comment by Fernando-Dourado** [comment]
trigger uts

**Review by rafaelmart1ns — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger ut0

**Comment by Fernando-Dourado** [comment]
trigger ut0

**Comment by Fernando-Dourado** [comment]
trigger uts

**Review by rafaelmart1ns — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger ut8

**Review by vcalasansh — APPROVED** [review]
> Reviewed 3 of 5 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52271)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions.

**Review by lasdox — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger uts_JDK11

**Review by jennyjames21 — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger review

**Comment by Fernando-Dourado** [comment]
trigger ut9

**Comment by Fernando-Dourado** [comment]
trigger ut9

**Pull request merged**